pkg/getty/README.md

```diff
@@ -10,7 +10,7 @@ If you want a console getty, add the following to your `moby.yml`:
 ```
 services:
   - name: getty
-    image: "linuxkit/getty:e36b5e794256c2a56c000035f73a4d46f7ffb3e6"
```

Comment:
Not quite, @ijc . The has is not the latest commit to the entire repo, just the latest commit that affects the `pkg/getty` directory. Look at the `HASH` code from the Makefile:

```
HASH?=$(shell git ls-tree HEAD -- ../$(notdir $(CURDIR)) | awk '{print $$3}')
```

It takes the head of the `pkg/getty/` dir. As long as you don't change that dir, nothing changes.

I still do find it annoying, though, e.g. if you change the `Makefile`, the hash _does_ change. Which is why, outside of this particular project (where there are consistent ways of doing it), I put everything that should go into an image into a subdir, so that only if stuff inside that dir changes, it affects the hash, but not the `README.md` or `Makefile`. So `pkg/getty` would look like:

```
pkg/
 |- getty/
          |- src/
                |- etc/
                      |- ....
                |- usr/
                      |- ....
          |- Makefile
          |- Dockerfile
          |- README.md
```

And the line in `Makefile` instead is:

```
HASH?=$(shell git ls-tree HEAD -- ./src/ | awk '{print $$3}')
```

Look at `projects/compose/` for an example of that.